docs/config.md

```diff
@@ -4,20 +4,20 @@
 
 ## Configuration file
 
-When a pipeline script is launched, Nextflow looks for configuration files in multiple locations. Since each configuration file can contain conflicting settings, the sources are ranked to determine which settings are applied. Possible configuration sources, in order of priority:
+When a pipeline script is launched, Nextflow looks for configuration files in multiple locations. Since each configuration file may contain conflicting settings, they are resolved as follows (from highest to lowest priority):
 
 1. Parameters specified on the command line (`--something value`)
 2. Parameters provided using the `-params-file` option
-3. Config file specified using the `-c my_config` option
-4. The config file named `nextflow.config` in the current directory
-5. The config file named `nextflow.config` in the workflow project directory
+3. Config file specified using the `-c <config-file>` option
+4. The config file `nextflow.config` in the launch directory
+5. The config file `nextflow.config` in the project directory
```

Comment:
After 4. and 5., I would add an item in this list as well:
```
6. The module config file `module.config` in the project module directory (`process` parameters only, see Module config(REF) section in the Modules(REF) page)
```

I reckon the module.config takes higher priority then the HOME config file, `$HOME/.nextflow/config`, right?

Reply:
It would be lower than the HOME config file, because the config files from 3-6 are resolved into a single "pipeline config" which takes precedence over module configs. I will add it as 7